### PR TITLE
feat: add @tsky/lex-cli for generating typescript definitions

### PR DESCRIPTION
**Pull request by skiniks**
This PR introduces `@tsky/lex-cli`, a TypeScript definition generator forked from `@atcute/lex-cli`.

Key changes:
- Created `@tsky/lex-cli` package based on `@atcute/lex-cli` v1.0.3
- Updated imports and package names to use `@tsky` namespace
- Switched from `@externdefs/collider` to `clipanion`
- Switched from `@badrap/valita` to `typanion` for schema validation 
- Simplified file structure by consolidating utility functions
- Added support for subscriptions in type generation
- Expanded JSDoc documentation handling

The PR includes generated type definitions for review in `packages/lexicons/src/lib/lexicons.ts`. Please review both the generator code and the resulting type definitions and let me know what you think!

Edit: Following suggestion from mary, replaced symbolic type branding with a more pragmatic approach using concrete `$type` fields. This change:

- Removes Brand namespace and symbolic type system
- Introduces TypedBase (optional `$type`) and RecordBase (required `$type`)
- Uses real runtime properties instead of nonexistent unique symbols
- Simplifies type unions with explicit type discrimination
- Improves TypeScript type inference and debugging

related to #30

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,37 +1,37 @@
-{
```
Comment (43081j) [inline package.json:1]: Presumably this was crlf to lf or vice versa? 

This reminds me, we should add an editor config 👀 #29

Comment (skiniks) [reply package.json:1]: good catch, i think that's exactly what this was!

File: packages/lexicons/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './src/index';
```
Comment (43081j) [inline packages/lexicons/index.ts:1]: In lex-cli, all imports have file extensions. Whereas in lexicons, they don't 

We should probably make lexicons also have extensions & in some future pr, lint for it

**Comment by anbraten** [comment]
@skiniks Could you merge main to resolve conflicts.

**Comment by skiniks** [comment]
@anbraten all set!

File: packages/lexicons/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
+import {
```
Comment (anbraten) [inline packages/lexicons/src/index.ts:1]: Haven't checked the complete `lexicons.ts` file 😅  However as I understand apart from these few  utilities in here, the whole package only has types and therefore will not add to the package size of our client, right?

**Review by anbraten — APPROVED** [review]

**Comment by anbraten** [comment]
In general thanks again for the amazing work 👍🏾 . This will be a good starting point to continue building the client functions now. Would suggest to merge this as is and adjust further when necessary while using the types.

File: packages/lexicons/src/index.ts (hunk 3 above)
Comment (skiniks) [reply packages/lexicons/src/index.ts:1]: exactly! only the type guards will be included—so very minimal impact on bundle size

**Pull request merged**